src/v/kafka/client/client.cc

```diff
@@ -144,6 +144,7 @@ ss::future<> client::apply(metadata_response res) {
 ss::future<> client::mitigate_error(std::exception_ptr ex) {
     return _external_mitigate(ex).handle_exception(
       [this](std::exception_ptr ex) {
+          _gate.check();
```

Comment:
if you move this into the try-block then it looks like you could then follow the path with some error logging?

```
          } catch (const ss::gate_closed_exception&) {
              vlog(kclog.debug, "gate_closed_exception");
```

if that is desired behavior